### ISSUE TITLE
[FeatureRequests] Dynamic product for SPM

### ISSUE THREAD

**Issue by GusevAndrey** [Alamofire/Alamofire#3347]
Hi

I'm developing a swift package and want using Alamofire in it.
The package itself is expected to be linked in other products that are using Alamofire too.

Unfortunately Alamofire swift package product is static so using both my package and Alamofire itself leads to duplicate symbols.

Looks like solution is pretty simple - add `Alamofire-dynamic` product to swift package manifest like so:

```swift
 products: [
  .library(name: "Alamofire",
                type: .static,
                targets: ["Alamofire"]),

   .library(name: "Alamofire-dynamic",
                 type: .dynamic,
                 targets: ["Alamofire"]),
],
```
This will allow to link Alamofire in product and it's dependencies while stay compatible to all current users of static Alamofire product.

**Comment by jshier** [comment]
Typically it's up to the SPM user to decide how to integrate Alamofire, just like it would be up to them to decide whether to use `Alamofire` or `Alamofire-dynamic`, so I'm not sure what this approach would actually provide. Even Xcode, which doesn't provide control over static or dynamic integration, usually tries to decide which it should use based on other dependencies. So can you provide some more specific examples of what this would actually fix and how?

**Comment by GusevAndrey** [comment]
Here I assembled a demo project which illustrates the problem: [MyAwsomProject.zip](https://github.com/Alamofire/Alamofire/files/5454395/MyAwsomProject.zip)

You can see, that test application and package's target (MyAwsomeTarget) are both linked against Alamofire (the static one) and build says following:

> Build system information
> error: Swift package product 'Alamofire' is linked as a static library by 'MyAwsomProject' and 'MyAwsomePackage-dynamic'. This will result in duplication of library code.

The problem can be solved if both - the app and the package would be use dynamic product: only one Alamofire symbols will be actually linked.

As example you can you my fork to link dynamically (both: the package **and** the app should use this dependency) and see the result:
```swift
.package(url: "https://github.com/GusevAndrey/Alamofire", .branch("master"))
```

[There is discussion on Apple's forum](https://forums.swift.org/t/swift-packages-in-multiple-targets-results-in-this-will-result-in-duplication-of-library-code-errors/34892/44
) about this problem which can provide some more information

**Comment by jshier** [comment]
Apple's suggested workaround for that error is for users to consume packages through a single local package which can control whether dependencies are static or dynamic. I'm reluctant to introduce any sort of packages changes that will only last as long as Xcode doesn't allow direct control over package linking. I'll give it some consideration though.

**Comment by jshier** [comment]
This issue should be resolved in Xcode 12.5, can you give it a try?

**Comment by GusevAndrey** [comment]
Hi, @jshier! Sorry for delay.

I can confirm that example project MyAwsomProject app is correctly builds on Xcode 12.5 (and does not on 12.4).
So, looks like nothing more is needed.
